CMakeLists.txt

```diff
@@ -30,6 +30,14 @@ set(CMAKE_CXX_STANDARD 11)
 
 enable_testing()
 
+# Disable complex math NaN/INFO range checking for performance
+include(CheckCXXCompilerFlag)
+check_cxx_compiler_flag(-fcx-limited-range HAVE_CX_LIMITED_RANGE)
```

Comment:
I would assume C and C++ compilers are basically the same piece of software (gcc/g++) but I'd still add an extra test for the C compiler.
Besides, this looks good.

Reply:
Good call.  Added separate check.